pkg/csi/service/common/commonco/k8sorchestrator/k8sorchestrator.go

```diff
@@ -353,3 +467,59 @@ func (c *K8sOrchestrator) IsFSSEnabled(ctx context.Context, featureName string)
 	log.Debugf("cluster flavor %q not recognised. Defaulting to false", c.clusterFlavor)
 	return false
 }
+
+// GetVolumeAnnotations fetches annotations from PVC bound to passed volumeID and returns
+// annotation key-value pairs as a map
+func (c *K8sOrchestrator) GetVolumeAnnotations(ctx context.Context, volumeID string) (map[string]string, error) {
+	log := logger.GetLogger(ctx)
+	log.Debugf("Getting volume annotations for volume: %s", volumeID)
+	if pvc, ok := c.volumeIDToPvcMap[volumeID]; ok {
+		parts := strings.Split(pvc, "/")
+		pvcNamespace := parts[0]
+		pvcName := parts[1]
+
+		pvcObj, err := c.k8sClient.CoreV1().PersistentVolumeClaims(pvcNamespace).Get(ctx, pvcName, metav1.GetOptions{})
+		if err != nil {
+			log.Errorf("error getting pvc %s/%s from api server: %v", pvcNamespace, pvcName, err)
+			return nil, err
+		}
+
+		return pvcObj.Annotations, nil
+	}
+
+	errMsg := fmt.Sprintf("could not find pvc for volumeID: %s", volumeID)
+	log.Debugf(errMsg)
+	return nil, errors.New(errMsg)
+}
+
+// UpdateVolumeAnnotations updates annotations passed as key-value pairs
+// on PVC bound to passed volumeID
+func (c *K8sOrchestrator) UpdateVolumeAnnotations(ctx context.Context, volumeID string, annotations map[string]string) error {
+	log := logger.GetLogger(ctx)
+	if pvc, ok := c.volumeIDToPvcMap[volumeID]; ok {
+		parts := strings.Split(pvc, "/")
+		pvcNamespace := parts[0]
+		pvcName := parts[1]
+
+		pvcObj, err := c.k8sClient.CoreV1().PersistentVolumeClaims(pvcNamespace).Get(ctx, pvcName, metav1.GetOptions{})
```

Comment:
Why are we not using PVCListers() in `c.informerManager` ? This will be a faster lookup instead of querying the api server

Reply:
That makes sense. Are there any potential sync issues while using PVCListers()?